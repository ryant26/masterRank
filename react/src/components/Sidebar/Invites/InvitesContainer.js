import React from 'react';
import InvitesCard from './InvitesCard';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const InvitesContainer = (props) => {
  return (
    <InvitesCard groupInvites={props.groupInvites}/>
  );
};

InvitesContainer.propTypes = {
  groupInvites: PropTypes.array.isRequired
};

const mapStateToProps = (state) => {
  return {
    groupInvites: state.groupInvites
  };
};

export default connect(mapStateToProps)(InvitesContainer);