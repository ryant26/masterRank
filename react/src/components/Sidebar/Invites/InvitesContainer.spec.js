import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import INVITES from 'resources/groupInvites';
import configureStore from 'redux-mock-store';
import {MemoryRouter} from 'react-router-dom';

import InvitesContainer from 'components/Sidebar/Invites/InvitesContainer';

const mockStore = configureStore();

const renderWithStore = (store) => {
    return renderer.create(
        <MemoryRouter>
            <InvitesContainer store={store}/>
        </MemoryRouter>
    );
};

describe('HeroCard Component',()=> {
    let store;

    beforeEach(() => {
        store = mockStore({groupInvites: INVITES});
    });

    it('should render without exploding', () => {
        const wrapper = mount(
            <MemoryRouter>
                <InvitesContainer store={store}/>
            </MemoryRouter>
        );

        const InvitesContainerComponent = wrapper.find(InvitesContainer);
        expect(InvitesContainerComponent).toBeTruthy();
    });

    it('should render a single invite', () => {
        store = mockStore({groupInvites: [INVITES[0]]});
        const component = renderWithStore(store);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render two invites', () => {
        store = mockStore({groupInvites: [INVITES[0], INVITES[1]]});
        const component = renderWithStore(store);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render three invites', () => {
        store = mockStore({groupInvites: [INVITES[0], INVITES[1], INVITES[2]]});
        const component = renderWithStore(store);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});