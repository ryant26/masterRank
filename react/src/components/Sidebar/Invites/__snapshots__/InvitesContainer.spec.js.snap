// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard Component should render a single invite 1`] = `
<div
  className="flex flex-column sidebar-card"
>
  <div
    className="flex align-center justify-between sidebar-title"
  >
    <div>
      Invites
    </div>
    <div
      className="Badge"
    >
      <div
        className="background flex justify-center align-center"
      >
        <div
          className="label"
        >
          3
        </div>
      </div>
    </div>
  </div>
  <div
    className="InvitesList"
  >
    <div
      className="list-container"
    >
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            0
            /
            3
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
    </div>
  </div>
  <div
    className="button-secondary flex align-center justify-center"
  >
    <div
      className="button-content"
    >
      All Invites
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should render three invites 1`] = `
<div
  className="flex flex-column sidebar-card"
>
  <div
    className="flex align-center justify-between sidebar-title"
  >
    <div>
      Invites
    </div>
    <div
      className="Badge"
    >
      <div
        className="background flex justify-center align-center"
      >
        <div
          className="label"
        >
          3
        </div>
      </div>
    </div>
  </div>
  <div
    className="InvitesList"
  >
    <div
      className="list-container"
    >
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            0
            /
            3
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
    </div>
  </div>
  <div
    className="button-secondary flex align-center justify-center"
  >
    <div
      className="button-content"
    >
      All Invites
    </div>
  </div>
</div>
`;

exports[`HeroCard Component should render two invites 1`] = `
<div
  className="flex flex-column sidebar-card"
>
  <div
    className="flex align-center justify-between sidebar-title"
  >
    <div>
      Invites
    </div>
    <div
      className="Badge"
    >
      <div
        className="background flex justify-center align-center"
      >
        <div
          className="label"
        >
          3
        </div>
      </div>
    </div>
  </div>
  <div
    className="InvitesList"
  >
    <div
      className="list-container"
    >
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            0
            /
            3
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
      <div
        className="InviteListItem flex flex-column justify-center"
      >
        <div
          className="flex justify-between group-details"
        >
          <div>
            luckbomb#1234
          </div>
          <div>
            1
            /
            4
          </div>
        </div>
        <div
          className="sub-title"
        >
          Just now
        </div>
      </div>
    </div>
  </div>
  <div
    className="button-secondary flex align-center justify-center"
  >
    <div
      className="button-content"
    >
      All Invites
    </div>
  </div>
</div>
`;
